tardis/tardis_portal/api_v2/serializers/experiment.py

```diff
@@ -0,0 +1,178 @@
+
+    class Meta:
+        model = Project
+        fields = [
+            "id",
+            "name",
+        ]
+        if (
+            "tardis.apps.identifiers" in settings.INSTALLED_APPS
+            and "project" in settings.OBJECTS_WITH_IDENTIFIERS
+        ):
+            fields.append("identifiers")
+
+
+# TODO: Add an ExperimentDatasetSerializer to provide more useful data than just the id
+
+
+class ExperimentSerializer(serializers.ModelSerializer):
+    experimentparameterset_set = ExperimentParameterSetSerializer(many=True)
+    created_by = UserSerializer(many=False)
+    user_acls = serializers.SerializerMethodField("get_user_acls")
+    group_acls = serializers.SerializerMethodField("get_group_acls")
+    projects = serializers.SerializerMethodField("get_projects")
+
+    # datasets = serializers.SerializerMethodField("get_datasets")
+
+    if (
+        "tardis.apps.identifiers" in settings.INSTALLED_APPS
+        and "experiment" in settings.OBJECTS_WITH_IDENTIFIERS
+    ):
+        identifiers = ExperimentIDSerializer(many=True)
+    if "tardis.apps.dataclassification" in settings.INSTALLED_APPS:
+        data_classification = ExperimentDataclassificationSerializer()
+
+    class Meta:
+        model = Experiment
+        fields = [
+            "title",
+            "description",
+            "start_time",
+            "end_time",
+            "created_time",
+            "update_time",
+            "created_by",
+            "datasets",
+            "experimentparameterset_set",
+            "url",
+            "user_acls",
+            "group_acls",
+        ]
+        if (
+            "tardis.apps.identifiers" in settings.INSTALLED_APPS
+            and "experiment" in settings.OBJECTS_WITH_IDENTIFIERS
+        ):
+            fields.append("identifiers")
+        if "tardis.apps.dataclassification" in settings.INSTALLED_APPS:
+            fields.append("data_classification")
+        if "tardis.apps.projects" in settings.INSTALLED_APPS:
+            fields.append("projects")
+
+    def get_user_acls(self, obj):  # TODO wrap in tests for micro/macro ACLS
+        acls = obj.experimentacl_set.select_related("user").filter(user__isnull=False)
+        return [
+            {
+                "user": acl.get_related_object().username,
+                "can_download": acl.canDownload,
+                "see_sensitive": acl.canSensitive,
+                "is_owner": acl.isOwner,
+            }
+            for acl in acls
+        ]
+
+    def get_group_acls(self, obj):  # TODO wrap in tests for micro/macro ACLS
+        acls = obj.experimentacl_set.select_related("group").filter(group__isnull=False)
+        return [
+            {
+                "group": acl.get_related_object().name,
+                "can_download": acl.canDownload,
+                "see_sensitive": acl.canSensitive,
+                "is_owner": acl.isOwner,
+            }
+            for acl in acls
+        ]
```

Comment:
For `Experiments`, `Datasets` and `Datafiles`, setting ACL value to `[]` is distinct from `None`. `[]` implies an override of its parent ACLs, whereas `None` means this object will inherit parent ACLs. I'm not quite familiar with this bit of Django code but it may be returning `[]` regardless of whether there is an overriding ACL applied to it. If this is so, it ultimately won't affect ingestion outcome, as it won't try to change ACL settings of already ingested objects. But it will look different on the wizard and the web portal.

Reply:
Good catch. I think you are right w.r.t. the outcome for the yaml_dump, but probably needs more thought for the serializers, particularly when POSTing/PATCHing. I'll add a TODO comment so we don't lose this